code/data_structures/src/list/doubly_linked_list/C/doubly_linked_list_menu_driven.c

```diff
@@ -0,0 +1,238 @@
+                tmp = (struct Node *)malloc(sizeof(struct Node));
+                tmp = HEAD;
+                printf("\tENTER NODE INDEX : ");
+                scanf("%d", &value);
+                while (--value)
+                {
+                    tmp = tmp->next;
+                }
+                new_node = tmp;
+                tmp->next = tmp->next->next;
+                new_node->next->previous = tmp;
+                printf("DELETED NODE ASSOCIATED TO %d", new_node->data);
+                tmp = NULL;
+                new_node = NULL;
+                free(tmp);
+                free(new_node);
+                break;
+            case 3:
+                tmp = (struct Node *)malloc(sizeof(struct Node));
+                tmp = LAST;
+                LAST = LAST->next;
+                printf("DELETED NODE ASSOCIATED TO %d", tmp->data);
+                tmp = NULL;
+                free(tmp);
+                break;
+            }
+            --size;
+            break;
+        case 3:
+            if (HEAD == NULL || LAST == NULL)
+            {
+                printf("UNDERFLOW\n");
+            }
+            else
+            {
+                tmp = (struct Node *)malloc(sizeof(struct Node));
+                tmp = HEAD;
+                while (tmp != NULL)
+                {
+                    printf("%d ", tmp->data);
+                    tmp = tmp->next;
+                }
+                tmp = NULL;
+                free(tmp);
+            }
+            break;
+        case 4:
+            tmp = (struct Node *)malloc(sizeof(struct Node));
+            new_node = (struct Node *)malloc(sizeof(struct Node));
+            tmp = HEAD;
+            while (tmp != NULL)
+            {
+                new_node = tmp->next;
+                tmp->next = tmp->previous;
+                tmp->previous = new_node;
+                tmp = new_node;
+            }
+            new_node = HEAD;
+            HEAD = LAST;
+            LAST = new_node;
+            new_node = NULL;
+            tmp = NULL;
+            free(tmp);
+            free(new_node);
+            break;
+        case 5:
+            printf("\tENTER NUMBER TO SEARCH : ");
+            scanf("%d", &value);
+            flag = 0;
+            itr = 0;
+            tmp = (struct Node *)malloc(sizeof(struct Node));
+            ;
+            tmp = HEAD;
+            while (tmp != NULL)
+            {
+                if (value == tmp->data)
+                {
+                    flag = 1;
+                    break;
+                }
+                tmp = tmp->next;
+                ++itr;
+            }
+
+            if (flag)
+            {
+                printf(
+                    "NUMBER FOUND AT %d NODE",
+                    itr + 1); // to print non index value
+            }
+            else
+            {
+                printf("NODE NOT FOUND");
+            }
+            tmp = NULL;
+            free(tmp);
```

Comment:
You're supposed to deallocate _before_ setting to `NULL`. Otherwise, you have memory leaks.